app/views/hosts/_operating_system.html.erb

```diff
@@ -1,6 +1,6 @@
 
 <%= select_f f, :architecture_id, accessible_resource(f.object, :architecture), :id, :to_label, {:include_blank => true},
-    {:onchange => 'architecture_selected(this);', :'data-url' => method_path('architecture_selected'), :'data-type' => controller_name.singularize,
+    {:onchange => 'architecture_selected(this);', :'data-url' => method_path('architecture_selected'), :'data-type' => "host",
```

Comment:
I'm not sure if that's the way to go, since we are also using this form for `hostgroups`.

Reply:
we are using https://github.com/theforeman/foreman/blob/develop/app/views/common/os_selection/_operatingsystem.html.erb for hostgroups

Reply:
Yeah, you are right we are not using this one, my bad.

Reply:
I believe this template is only used in the host form (which includes VMs as well). I'm considering whether we should just pass `data-type` as `host`, or if it would be clearer and more consistent with the rest of the code to add a specific check for this scenario.

Reply:
We can override the `controller_type` for the compute_resources_vm controller but I guess that could maybe break things.

Reply:
I wouldn't modify the `controller_name`, but I'm wondering whether hard-coding it on the frontend is the best approach in this case.

Reply:
I believe we have no other option in this case!
I think it shall most probably not impact other use cases but I'm open to any alternative approaches anyone may have!

Reply:
Since all other forms use the same syntax (e.g., `'data-type' => controller_name.singularize`), I suggest we only handle the special case where the controller is `ComputeResourcesVms`.

Reply:
That's exactly what we are doing, we can override it as well but its just the same approach

Reply:
Oh that sounds good. I'll update it

Reply:
@girijaasoni can you please update this line so can merge this?